src/api/forms/service.js

```diff
@@ -326,6 +326,45 @@ export async function createDraftFromLive(formId, author) {
   }
 }
 
+/**
+ * Deletes a form (metadata and definition)
+ * @param {string} formId
+ * @param {boolean} force - deletes the form even if it's live, and ignores failures to delete the form definition.
+ */
+export async function deleteForm(formId, force = false) {
+  logger.info(`Deleting form with ID ${formId}`)
+
+  const form = await getForm(formId)
+
+  if (!force && form.live) {
+    throw Boom.badRequest(
+      `Form with ID '${formId}' is live and cannot be deleted. Set force=true to delete the form anyway.`
+    )
+  }
+
+  const session = client.startSession()
+
+  try {
+    await session.withTransaction(async () => {
+      await formMetadata.drop(formId, session)
+
+      try {
+        await formDefinition.drop(formId, session)
+      } catch (err) {
+        // we might have old forms that don't have form definitions but do have metadata entries.
+        // TODO keep this as a short term only, then remove once cleaned up.
+        if (!force) {
+          throw err
+        }
```

Comment:
I don't particularly like this, but I intend to remove the "if force, ignore code" once we've cleaned up our environments.